dlib/interfaces/map_pair.h

```diff
@@ -55,11 +55,6 @@ namespace dlib
                 - returns a non-const reference to the value associated with key
         !*/
 
-    protected:
-
-        // restricted functions
-        map_pair<T1,T2>& operator=(const map_pair<T1,T2>&) {return *this;} // no assignment operator
```

Comment:
Still trying to figure out how exactly this one should be treated.

Reply:
Let's discuss this here: https://github.com/davisking/dlib/pull/2490#pullrequestreview-873221071